cmd/edgectl/aes_install.go

```diff
@@ -535,124 +563,122 @@ func (i *Installer) Perform(kcontext string) Result {
 
-	// Try to determine cluster type from node labels
-	isKnownLocalCluster := false
-	if clusterNodeLabels, err := i.CaptureKubectl("get node labels", "", "get", "no", "-Lkubernetes.io/hostname"); err == nil {
-		clusterInfo := "unknown"
-		if strings.Contains(clusterNodeLabels, "docker-desktop") {
-			clusterInfo = "docker-desktop"
-			isKnownLocalCluster = true
-		} else if strings.Contains(clusterNodeLabels, "minikube") {
-			clusterInfo = "minikube"
-			isKnownLocalCluster = true
-		} else if strings.Contains(clusterNodeLabels, "kind") {
-			clusterInfo = "kind"
-			isKnownLocalCluster = true
-		} else if strings.Contains(clusterNodeLabels, "gke") {
-			clusterInfo = "gke"
-		} else if strings.Contains(clusterNodeLabels, "aks") {
-			clusterInfo = "aks"
-		} else if strings.Contains(clusterNodeLabels, "compute") {
-			clusterInfo = "eks"
-		} else if strings.Contains(clusterNodeLabels, "ec2") {
-			clusterInfo = "ec2"
-		}
-		i.SetMetadatum("Cluster Info", "cluster_info", clusterInfo)
+	if err := chartDown.Download(); err != nil {
+		i.Report("fail_no_internet", ScoutMeta{"err", err.Error()})
+		return UnhandledErrResult(errors.Wrap(err, "download AES CRD manifests"))
 	}
 
-	// Find existing AES CRDs
-	aesCrds, err := i.GetAESCRDs()
+	chartRel, err := chartDown.GetReleaseMgr(defInstallNamespace, defChartValues)
+	defer func() { _ = chartDown.Cleanup() }()
+
+	// the AES version we have downloaded
+	i.version = strings.Trim(chartDown.GetChart().AppVersion, "\n")
+
+	// Try to verify the existence of an Ambassador deployment in the Ambassador
+	// namespace.
+	installedVersion, installedFamily, err := i.GetExistingInstallation()
 	if err != nil {
-		i.show.Println("Failed to get existing CRDs:", err)
-		aesCrds = []string{}
-		// Things will likely fail when we try to apply CRDs
-	}
-
-	// Figure out whether we need to apply the manifests
-	// TODO: Parse the downloaded manifests and look for specific CRDs.
-	// Installed CRDs may be out of date or incomplete (e.g., if there's an
-	// OSS installation present).
-	alreadyApplied := false
-	if len(aesCrds) > 0 {
-		// AES CRDs exist so there is likely an existing installation. Try to
-		// verify the existence of an Ambassador deployment in the Ambassador
-		// namespace.
-		installedVersion, err := i.GetInstalledImageVersion()
+		i.show.Println("Failed to look for an existing installation:", err)
+		installedVersion = "" // Things will likely fail when we try to apply manifests
+	}
+
+	alreadyInstalled := false
+	if installedVersion != "" {
+		alreadyInstalled = true
+	} else {
 		if err != nil {
-			i.show.Println("Failed to look for an existing installation:", err)
-			installedVersion = ""
-			// Things will likely fail when we try to apply manifests
+			i.log.Printf("when obtaining the chart manager: %s", err)
+			i.Report("fail_download", ScoutMeta{"err", err.Error()})
+			return UnhandledErrResult(err)
 		}
```

Comment:
Shouldn't we do this higher up, where we set `installedVersion = ""`?

Reply:
Damn, this is a merge error. Fixing.